extensions/panache/hibernate-orm-panache-common/runtime/src/main/java/io/quarkus/hibernate/orm/panache/common/runtime/CommonPanacheQueryImpl.java

```diff
@@ -95,13 +95,20 @@ private CommonPanacheQueryImpl(CommonPanacheQueryImpl<?> previousQuery, String n
             String parameterName;
             if (parameter.isAnnotationPresent(ProjectedFieldName.class)) {
                 final String name = parameter.getAnnotation(ProjectedFieldName.class).value();
-                if (name.isEmpty())
+                if (name.isEmpty()) {
                     throw new PanacheQueryException("The annotation ProjectedFieldName must have a non-empty value.");
+                }
+                parameterName = name;
+            } else if (parameter.isAnnotationPresent(io.quarkus.hibernate.orm.panache.ProjectedFieldName.class)) {
+                final String name = parameter.getAnnotation(io.quarkus.hibernate.orm.panache.ProjectedFieldName.class).value();
+                if (name.isEmpty()) {
+                    throw new PanacheQueryException("The annotation ProjectedFieldName must have a non-empty value.");
+                }
```

Comment:
You could have squashed these two if on the same condition using an ||, this would avoid the small code duplication.

Reply:
Well not really as I get the values from a different annotation. I think it's good enough as it is given we will drop one of the branch in 2.2.